src/EnergyPlus/PoweredInductionUnits.cc

```diff
@@ -285,23 +285,22 @@ void GetPIUs(EnergyPlusData &state)
                                                                  state.dataIPShortCut->cAlphaFieldNames,
                                                                  state.dataIPShortCut->cNumericFieldNames);
 
-        PIUNum = PIUIndex;
         GlobalNames::VerifyUniqueInterObjectName(state,
                                                  state.dataPowerInductionUnits->PiuUniqueNames,
                                                  state.dataIPShortCut->cAlphaArgs(1),
                                                  cCurrentModuleObject,
                                                  state.dataIPShortCut->cAlphaFieldNames(1),
                                                  ErrorsFound);
-        state.dataPowerInductionUnits->PIU(PIUNum).Name = state.dataIPShortCut->cAlphaArgs(1);
-        state.dataPowerInductionUnits->PIU(PIUNum).UnitType = cCurrentModuleObject;
-        state.dataPowerInductionUnits->PIU(PIUNum).UnitType_Num = DataDefineEquip::ZnAirLoopEquipType::SingleDuct_SeriesPIU_Reheat;
-        state.dataPowerInductionUnits->PIU(PIUNum).Sched = state.dataIPShortCut->cAlphaArgs(2);
+        auto &thisPIU = state.dataPowerInductionUnits->PIU(PIUIndex);
+        thisPIU.Name = state.dataIPShortCut->cAlphaArgs(1);
+        thisPIU.UnitType = cCurrentModuleObject;
+        thisPIU.UnitType_Num = DataDefineEquip::ZnAirLoopEquipType::SingleDuct_SeriesPIU_Reheat;
+        thisPIU.Sched = state.dataIPShortCut->cAlphaArgs(2);
```

Comment:
The rest of changes are about taking references and limiting scope of variables (and initializing them)